File: src/test/programmers/python/level2/FibonacciNumber.py
d = [-1] * 100001
d[0] = 0
d[1] = 1
def solution(n):
    for i in range(2, n + 1):
        d[i] = d[i - 1] + d[i - 2]
    return d[n] % 1234567

print(solution(3))

File: src/test/programmers/java/level2/FibonacciNumber.java
package test.programmers.java.level2;

public class FibonacciNumber {
    public static void main(String[] args) {
        System.out.println(solution(4));
    }

    static int solution(int n) {
        int answer = 0;

        int ppnum = 1;
        int pnum = 1;
        int divisor = 1234567;

        if(n == 2) return 1;

        for(int i = 3; i <= n; i++) {
            answer = (ppnum + pnum) % divisor;
            ppnum = pnum;
            pnum = answer;
        }

        return answer;
    }
}
